extensions/shared/videopress-plan-check.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { __, _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './videopress-plan-check.scss';
+import UpgradeNudge from './components/upgrade-nudge';
+import getJetpackExtensionAvailability from './get-jetpack-extension-availability';
+//import isSimpleSite from './site-type-utils';
+
+const replaceMediaPlaceholder = createHigherOrderComponent(
+	OriginalPlaceholder => props => {
+		const { available } = getJetpackExtensionAvailability( 'videopress' );
+
+		if (
+			//! isSimpleSite || // Only show nudge on dotcom simple sites.
+			'wp-block-video' !== props.className || // Only show nudge on core video blocks.
```

Comment:
This breaks if the user adds a custom class to the block (got caught by this on google calendar block) as className prop becomes a space separated string of all the class names in this instance.

<img width="892" alt="Screen Shot 2020-02-21 at 12 38 32 PM" src="https://user-images.githubusercontent.com/3629020/74990139-4d37de00-54a7-11ea-980b-557edcdcfc8c.png">

<img width="815" alt="Screen Shot 2020-02-21 at 12 38 41 PM" src="https://user-images.githubusercontent.com/3629020/74990137-49a45700-54a7-11ea-8e19-eea9bfe063c2.png">

It is possible to get just the base class name like this https://github.com/Automattic/jetpack/pull/13999/files#diff-79dff83b7f1000a7b0f61a8752aa35cfR113

Reply:
... just saw comment above that `name` prop is not available, so the suggested call to get the default class name won't work as this requries `name` 🤦

Reply:
Had a quick look and there don't seem to be many other options with the props available, but you could do:

```
props.className.indexOf( 'wp-block-video' ) !== -1;
```
or
```
props.accept === 'video/*'
```

Reply:
Using @pablinos's tip below and the `blocks.registerBlockType` filter, I can get the block name prop, and then use that to limit the `editor.MediaPlaceholder` filter. Refactor coming up. 👍

Reply:
Looking at this again now the refactor is in, I still need to check inside the filter function despite it only being applied after a block name check in the `blocks.registerBlockType` filter. The filter is still broadly applied to every block.